desertbot/modules/utils/WebUtils.py

```diff
@@ -41,32 +41,39 @@ def onLoad(self):
                        "application/rss+xml, application/atom+xml, application/rdf+xml, "
                        "application/json")
 
-    def isPublicURL(self, url: str) -> bool:
-        parsed = urlparse(url)
-        host = socket.gethostbyname(parsed.hostname)
-        ip = ipaddress.ip_address(host)
-        return ip.is_global
-
     def fetchURL(self, url: str,
                  params: Any=None,
                  extraHeaders: Optional[Dict[str, str]]=None) -> Optional[Response]:
+
+        # stop changing DNS attacks by locking in the IP
+        origURL = url
+        parsedURL = urlparse(url)
+        host = socket.gethostbyname(parsedURL.hostname)
+        ip = ipaddress.ip_address(host)
+        port = '' if parsedURL.port is None else f':{parsedURL.port}'
+        url = urlunparse(parsedURL._replace(netloc=f"{ip}{port}"))
```

Comment:
Maybe this should be a function rather than being copy pasted.

Reply:
Returning (url, host, ip) I suppose, since we need all 3.
Does anyone have good function name ideas?

Reply:
lockURLtoIP? lockHostIP? Could even do like, processURL and have a docstring that explains the function...

Reply:
Having a function that returns three strings feels a bit janky to me, but I don't know a better way to handle it

Reply:
Yeah, it's either return 3 things, or have 3 different one or two line functions that may as well not be functions.

Anyway, done.